smoke/LeafDevice/details/Details.cs

```diff
@@ -50,48 +56,62 @@ bool useWebSockets
             {
                 this.serviceClientTransportType = ServiceClientTransportType.Amqp_WebSocket_Only;
                 this.eventHubClientTransportType = EventHubClientTransportType.AmqpWebSockets;
-                this.deviceClientTransportType = DeviceClientTransportType.Mqtt_WebSocket_Only;
+                this.deviceTransportSettings = new ITransportSettings[] { new MqttTransportSettings(DeviceClientTransportType.Mqtt_WebSocket_Only) };
             }
             else
             {
                 this.serviceClientTransportType = ServiceClientTransportType.Amqp;
                 this.eventHubClientTransportType = EventHubClientTransportType.Amqp;
-                this.deviceClientTransportType = DeviceClientTransportType.Mqtt;
+                this.deviceTransportSettings = new ITransportSettings[] { new MqttTransportSettings(DeviceClientTransportType.Mqtt_Tcp_Only) };
             }
         }
 
         protected Task InstallCaCertificate()
         {
-            var store = new X509Store(StoreName.Root, StoreLocation.CurrentUser);
-            store.Open(OpenFlags.ReadWrite);
-            store.Add(new X509Certificate2(X509Certificate.CreateFromCertFile(this.certificateFileName)));
-            store.Close();
+            // Since Windows will pop up security warning when add certificate to current user store location;
+            // Therefore we will use CustomCertificateValidator instead.
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                var store = new X509Store(StoreName.Root, StoreLocation.CurrentUser);
+                store.Open(OpenFlags.ReadWrite);
+                store.Add(this.GetCertificate());
+                store.Close();
+            }
+
             return Task.CompletedTask;
         }
 
+        X509Certificate2 GetCertificate() => new X509Certificate2(X509Certificate.CreateFromCertFile(this.certificateFileName));
+
         protected async Task ConnectToEdgeAndSendData()
         {
-            Microsoft.Azure.Devices.IotHubConnectionStringBuilder builder = Microsoft.Azure.Devices.IotHubConnectionStringBuilder.Create(this.iothubConnectionString);
+            IotHubConnectionStringBuilder builder = IotHubConnectionStringBuilder.Create(this.iothubConnectionString);
             string leafDeviceConnectionString = $"HostName={builder.HostName};DeviceId={this.deviceId};SharedAccessKey={this.context.Device.Authentication.SymmetricKey.PrimaryKey};GatewayHostName={this.edgeHostName}";
-            
-            this.context.DeviceClientInstance = Option.Some(DeviceClient.CreateFromConnectionString(leafDeviceConnectionString, this.deviceClientTransportType));
+
+            // Need to use CustomCertificateValidator since we can't automate to install certificate on Windows
+            // For details, refer to InstallCaCertificate method
```

Comment:
This logic in 2 places is confusing..
I would say refactor InstallCaCertificate to something like InitializeServerCerts and in that have the if windows else logic..

Reply:
Updated in another PR.